v0.21: Facets and filters (#1041)
meilisearch/documentation: reference/api/filterable_attributes.md

# Filterable attributes

_Child route of the [settings route](/reference/api/settings.md)._

Attributes that can be used as filters for filtering and faceted search. [You can learn more about filtering and faceted search in our dedicated guide.](/reference/features/filtering_and_faceted_search.md)

Filterable attributes can also be updated through the [global settings route](/reference/api/settings.md#update-settings).

## Get filterable attributes

<RouteHighlighter method="GET" route="/indexes/:index_uid/settings/filterable-attributes" />

Get an index's [`filterableAttributes`](/reference/features/filtering_and_faceted_search.md).

### Path variables

| Variable      | Description   |
| ------------- | ------------- |
| **index_uid** | The index UID |

### Example

<CodeSamples id="get_filterable_attributes_1" />

#### Response: `200 Ok`

List the settings.

```json
["genres", "director"]
```

## Update filterable attributes

<RouteHighlighter method="POST" route="/indexes/:index_uid/settings/filterable-attributes" />

> Update an index's [filterable attributes list](/reference/features/filtering_and_faceted_search.md). This is a slow process that will take as much time as indexing the dataset.

Reviewer: Thanks @guimachiavelli, maybe we should also inform the user that he can specify them before indexing the documents if he already knows the fields on which he will want to filter or facet. This way, it avoids having to wait for this second automatic indexing phase.

Reviewer 2: Unfortunately this is not explained anywhere in the doc (unless I m mistaken). Maybe we can rephrase it 

```suggestion
Update an index's [filterable attributes list](/reference/features/filtering_and_faceted_search.md) re-indexes all documents in the index.  We suggest adding the settings after index creation, thus before adding your first documents.
```

Reviewer 2: I did not know lazy index also worked on `POST /settings` 🤯

Author: I prefer the shorter version, with minor cosmetic changes:

```suggestion
Update an index's [filterable attributes list](/reference/features/filtering_and_faceted_search.md). This will re-index all documents in the index.
```
